test/hotspot/jtreg/runtime/cds/appcds/aotCache/AOTCacheMXBeanTest.java

```diff
@@ -0,0 +1,154 @@
+            return new String[] {
+               "-Xlog:cds+class=trace",
+                "--add-modules=jdk.management"
+            };
+        }
+
+        @Override
+        public String[] appCommandLine(RunMode runMode) {
+            return new String[] {
+                mainClass, runMode.name()
+            };
+        }
+
+        @Override
+        public void checkExecution(OutputAnalyzer out, RunMode runMode) {
+            var name = runMode.name();
+            if (runMode.isApplicationExecuted()) {
+                out.shouldContain("Hello Leyden " + name);
+                if(runMode == RunMode.TRAINING) {
+                    if (false) {
+                        out.shouldContain("AOTMode = auto");
+                    } else {
+                        out.shouldContain("AOTMode = record");
+                    }
+                    out.shouldContain("Confirmed is recording");
+                    out.shouldContain("Confirmed recording duration > 0");
+                    out.shouldContain("Stopped recording successfully after an additional 10ms");
+                    out.shouldContain("Last recording duration > than previous duration");
+                    out.shouldContain("Confirmed recording stopped");
+                    out.shouldContain("Confirmed recording duration has not changed after 10ms");
+                } else if (runMode == RunMode.ASSEMBLY) {
+                    out.shouldNotContain("Hello Leyden ");
+                } else if (runMode == RunMode.PRODUCTION) {
+                    if (false) {
+                        out.shouldContain("AOTMode = auto");
+                    } else {
+                        out.shouldContain("AOTMode = on");
+                    }
+                    out.shouldContain("Confirmed is not recording");
+                    out.shouldContain("Confirmed recording duration == 0");
+                }
+                out.shouldNotContain("Thread interrupted");
+                out.shouldNotContain("Failed to stop recording");
+            }
+        }
+    }
+}
+
+class AOTCacheMXBeanApp {
+    public static void main(String[] args) {
+        System.out.println("Hello Leyden " + args[0]);
+        var aotBean = ManagementFactory.getPlatformMXBean(AOTCacheMXBean.class);
+        if (aotBean == null) {
+            System.out.println("AOTCacheMXBean is not available");
+            return;
+        }
+        System.out.println("AOTMode = " + aotBean.getMode());
+        if (aotBean.isRecording()) {
+            try {
+                System.out.println("Confirmed is recording");
+                var initialDuration = aotBean.getRecordingDuration();
+                System.out.println("Confirmed recording duration > 0");
+                Thread.sleep(10);
+                if (aotBean.endRecording()) {
+                    System.out.println("Stopped recording successfully after an additional 10ms");
+                    if (!aotBean.isRecording()) {
```

Comment:
if `aotBean.endRecording()` returns true, in what situation would `aotBean.isRecording()` return true?